src/dolphin/workflows/config/_unwrap_options.py

```diff
@@ -0,0 +1,255 @@
+    "SnaphuOptions",
+    "TophuOptions",
+    "SpurtOptions",
+]
+
+
+def to_tuple(v: int | tuple[int, int] | None) -> tuple[int, int]:
+    """Convert the input value to a tuple of two integers.
+
+    Parameters
+    ----------
+    v : Union[int, Tuple[int, int], None]
+        The value to be converted.
+
+    Returns
+    -------
+    Tuple[int, int]
+        Tuple containing two integers.
+
+    Examples
+    --------
+    >>> to_tuple(3)
+    (3, 3)
+    >>> to_tuple((2, 4))
+    (2, 4)
+    >>> to_tuple(None)
+    (1, 1)
+
+    """
+    if v is None:
+        return (1, 1)
+    if isinstance(v, int):
+        return (v, v)
+    return v
+
+
+class PreprocessOptions(BaseModel, extra="forbid"):
+    alpha: float = Field(
+        0.5,
+        description=(
+            "(for Goldstein filtering) Power parameter for Goldstein algorithm."
+        ),
+    )
+    max_radius: int = Field(
+        51,
+        ge=0.0,
+        description=("(for interpolation) maximum radius to find scatterers."),
+    )
+    interpolation_cor_threshold: float = Field(
+        0.5,
+        description=" Threshold on the correlation raster to use for interpolation. "
+        "Pixels with less than this value are replaced by a weighted "
+        "combination of neighboring pixels.",
+        ge=0.0,
+        le=1.0,
+    )
+
+
+class SnaphuOptions(BaseModel, extra="forbid"):
+    ntiles: tuple[int, int] = Field(
+        (1, 1),
+        description=(
+            "Number of tiles to split the inputs into using SNAPHU's internal tiling"
+        ),
+    )
+    tile_overlap: tuple[int, int] = Field(
+        (0, 0),
+        description=(
+            "Amount of tile overlap (in pixels) along the (row, col) directions."
+        ),
+    )
+    n_parallel_tiles: int = Field(
+        1,
+        description="Number of tiles to unwrap in parallel for each interferogram.",
+    )
+    init_method: Literal["mcf", "mst"] = Field(
+        "mcf",
+        description="Initialization method for SNAPHU.",
+    )
+    cost: Literal["defo", "smooth"] = Field(
+        "smooth",
+        description="Statistical cost mode method for SNAPHU.",
+    )
+
+
+class TophuOptions(BaseModel, extra="forbid"):
+    ntiles: tuple[int, int] = Field(
+        (1, 1),
+        description="Number of tiles to split the inputs into",
+    )
+    downsample_factor: tuple[int, int] = Field(
+        (1, 1),
+        description="Extra multilook factor to use for the coarse unwrap.",
+    )
+    init_method: Literal["mcf", "mst"] = Field(
+        "mcf",
+        description="Initialization method for SNAPHU.",
+    )
+    cost: Literal["defo", "smooth"] = Field(
+        "smooth",
+        description="Statistical cost mode method for SNAPHU.",
+    )
+
+    _to_tuple = field_validator("ntiles", "downsample_factor", mode="before")(to_tuple)
+
+
+class SpurtGeneralSettings(BaseModel):
+    use_tiles: bool = Field(default=True, description="Tile up data spatially.")
```

Comment:
Will we always use the density-based tiler? Should there be an option to select which tiling algorithm?

EDIT: Hmm I guess it would be complicated to support multiple tilers since different tiling algorithms may have different configuration parameters. Feel free to ignore.